cashctrl_ledger/ledger.py

```diff
@@ -976,6 +1085,21 @@ def base_currency(self) -> str:
         else:
             raise ValueError("Multiple base currencies defined.")
 
+    @base_currency.setter
+    def base_currency(self, currency):
```

Comment:
Yes, please do provided the workload is not excessive.

(Any restriction that we do not need to track or communicate makes the code easier to use by others.)

Reply:
Initially, I thought it would be a good idea to create the currency directly in the setter. However, after trying it out, I changed my mind. Here's an example of the code I was working with:

```python
@base_currency.setter
def base_currency(self, currency):
    currencies = self._client.list_currencies()
    if currency not in set(currencies["code"]):
        payload = {
            "code": currency,
            "isDefault": True,
            "description": "Some description needs to be added",
            "rate": "What rate should be?"
        }
        self._client.post("currency/update.json", data=payload)
    else:
        target_currency = currencies[currencies["code"] == currency].iloc[0]
        payload = {
            "code": currency,
            "id": target_currency["id"],
            "isDefault": True,
            "description": target_currency["description"],
            "rate": target_currency["rate"]
        }

    self._client.invalidate_currencies_cache()
```

We need to specify both the description and, more importantly, the rate. These parameters are optional for CashCtrl but crucial for our system. This introduces complications when using a setter. My proposal here is to raise an error if the currency has not yet been created.

Reply:
As discussed on our call this morning, the rate can be set to 1.0.

- The exchange 'rate' to translate any amount to the reporting currency is by definition 1.
- The description can be 'Reporting Currency' or similar.